libs/shared/ui/src/lib/components/status-menu/status-menu.tsx

```diff
@@ -0,0 +1,112 @@
+  RUNNING = 'Running',
+  STOPPED = 'Stopped',
+  ERROR = 'Error',
+  STARTING = 'Starting',
+  STOPPING = 'Stopping',
+}
+
+export interface StatusMenuProps {
+  status: StatusMenuState
+  menus?: { items: MenuItemProps[] }[]
+}
+
+export function StatusMenu(props: StatusMenuProps) {
+  const { status = StatusMenuState.RUNNING, menus = [] } = props
+  const [containerClass, setContainerClass] = useState('')
+  const [hoverClass, setHoverClass] = useState('')
+  const [focusClass, setFocusClass] = useState('')
+
+  useEffect(() => {
+    containerClassName()
+  }, [])
+
+  const containerClassName = () => {
+    switch (status) {
+      case StatusMenuState.RUNNING:
+        setContainerClass('bg-success-50 border-success-500 text-success-500')
+        setHoverClass('hover:bg-success-100')
+        break
+      case StatusMenuState.STOPPED:
+        setContainerClass('bg-element-light-lighter-300 border-element-light-lighter-700 text-text-400')
+        setHoverClass('hover:bg-element-light-lighter-400')
+        break
+      case StatusMenuState.ERROR:
+        setContainerClass('bg-error-50 border-error-400 text-error-500')
+        setHoverClass('hover:bg-error-100')
+        break
+      case StatusMenuState.STARTING || StatusMenuState.STOPPING:
+        setContainerClass('bg-progressing-50 border-progressing-400 text-progressing-500')
+        setHoverClass('hover:bg-progressing-100')
+        break
+      default:
+        setContainerClass('bg-success-50 border-success-500 text-success-500')
+        setHoverClass('hover:bg-success-100')
+    }
+  }
+
+  const focusStatus = (open: boolean) => {
+    if (open) {
+      switch (status) {
+        case StatusMenuState.RUNNING:
+          setFocusClass('bg-success-100')
+          break
+        case StatusMenuState.STOPPED:
+          setFocusClass('bg-element-light-lighter-400')
+          break
+        case StatusMenuState.ERROR:
+          setFocusClass('bg-error-100')
+          break
+        case StatusMenuState.STARTING || StatusMenuState.STOPPING:
+          setFocusClass('bg-progressing-100')
+          break
+        default:
+          setFocusClass('bg-success-100')
+      }
+    } else {
+      setFocusClass('')
+    }
+  }
+
+  const iconStatus = () => {
+    switch (status) {
+      case StatusMenuState.RUNNING:
+        return <Icon name="icon-solid-play" className="text-xs" />
+      case StatusMenuState.STOPPED:
+        return <Icon name="icon-solid-circle-stop" className="text-xs" />
+      case StatusMenuState.ERROR:
+        return <Icon name="icon-solid-circle-exclamation" className="text-xs" />
+      case StatusMenuState.STARTING || StatusMenuState.STOPPING:
+        return <Icon name="icon-solid-play" className="text-xs" />
+      default:
+        return <Icon name="icon-solid-play" className="text-xs" />
+    }
+  }
+
+  return (
+    <div className={`h-6 inline-flex items-center pl-2 border rounded ${containerClass}`} data-testid="statusmenu">
```

Comment:
I think the manage of colors is too complex, you can:
- Adding classes example `.status-menu .status-menu--error` with global colors and in the child of this classes you can add custom colors (example: `.status-menu--error { .icon { color: red } }` 
- The storybook looks buggy can you fix that?
- The buttons is cropped
<img width="359" alt="Capture d’écran 2022-04-25 à 17 02 48" src="https://user-images.githubusercontent.com/533928/165116827-4bf2c3e8-cd7d-48ef-aca6-059247b21a63.png">